graphql-relay/pkg/graph/pagination/pagination.go

```diff
@@ -26,15 +34,23 @@ func DecodeGraphqlID(encodedGraphqlID string) (NodeType, string, error) {
 	return NodeType(result[0]), result[1], nil
 }
 
-func NewConnection(nodes []node, hasNextPage, hasPreviousPage bool) *model.Connection {
+func toGraphqlModels[T Node](nodes []T) []model.Node {
+	res := make([]model.Node, 0, len(nodes))
+	for _, n := range nodes {
+		res = append(res, n.ToGraphqlNode())
+	}
+	return res
+}
+
+func NewConnection[T Node](nodes []T, hasNextPage, hasPreviousPage bool) *model.Connection {
 	return &model.Connection{
-		Edges:    NewEdges(nodes),
-		Nodes:    ToGraphqlModels(nodes),
-		PageInfo: NewPageInfo(nodes, hasNextPage, hasPreviousPage),
+		Edges:    newEdges(nodes),
+		Nodes:    toGraphqlModels(nodes),
+		PageInfo: newPageInfo(nodes, hasNextPage, hasPreviousPage),
 	}
 }
 
-func NewEdges(nodes []node) []*model.Edge {
+func newEdges[T Node](nodes []T) []*model.Edge {
```

Comment:
これってなんでジェネリクスで書いた方がいいのでしょうか？

Reply:
`NewConnection` accepts the type of generics, and because of this, it must also be used in this method.
If this type is `Node`, it will result in a type mismatch.